src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -415,8 +415,9 @@ private Task SendFrameLockAcquiredNonCancelableAsync(MessageOpcode opcode, bool
             }
             catch (Exception exc)
             {
-                return Task.FromException(_state == WebSocketState.Aborted ?
-                    CreateOperationCanceledException(exc) :
+                return Task.FromException(
+                    exc is OperationCanceledException ? exc :
+                    _state == WebSocketState.Aborted ? CreateOperationCanceledException(exc) :
```

Comment:
This seems a little difficult for me to parse without parens or using an actual `if`. But it's your code style, so I'm not really worried either way ;)